### PR TITLE
ramips: add support for D-Link DIR-3040 A1

### PR DESCRIPTION
**Pull request by Sinbiosis**
This adds support for the A1 hardware revision of the DIR-3040.
It is an exact copy of the [DIR-3060](https://wikidevi.wi-cat.ru/D-Link_DIR-3060_rev_A1) save for some cosmetic changes to the housing. Even going so far as having the same FCC ID. 

### Hardware specification:
SoC: MediaTek MT7621AT
Flash: Winbond W29N01HVSINA 128MB
RAM: Micron MT41K128M16JT-125 256MB
Ethernet: 5x 10/100/1000 Mbps
WiFi1: MT7615DN 2.4GHz N 2x2:2
WiFi2: MT7615DN 5GHz AC 2x2:2
WiFi3: MT7615N 5GHz AC 4x4:4
Button: WPS, Reset

### Flash instructions:
OpenWrt can be installed via [D-Link Recovery GUI](https://openwrt.org/docs/guide-user/installation/installation_methods/d-link_recovery_gui):
NOTE: Seems to only work in Firefox on Windows. Tried with Chrome on Windows, Firefox in Linux, and Chromium in Linux. None of these other browsers worked.

1. Push and hold reset button (on the bottom of the device) until power led starts flashing (about 10 secs or so) while plugging in the power cable.
2. Give it ~30 seconds, to boot the recovery mode GUI
3. Connect your client computer to LAN1 of the device
4. Set your client IP address manually to 192.168.0.2 / 255.255.255.0.
5. Call the recovery page for the device at http://192.168.0.1/
6. Use the provided emergency web GUI to upload and flash a new firmware to the device

Thanks to @Lucky1openwrt  and @iivailo for creating the DIR-3060 DTS file and related changes, so it was possible for me to adapt them to the DIR-3040, build images, test and fix minor issues.

### MAC Addresses:

| use | address | example |
| --- | --- | --- |
| LAN | label | f4:*:61 |
| WAN | label + 4 | f4:*:65 |
| WI1/2g | label + 2 | f4:*:62 |
| WI1/5g | label + 1 | f4:*:62 |
| WI2/5g | label + 3 | f4:*:64 |

The label MAC address was found in Factory, 0xe000

### Checklist:

- [x] nand
- [x] ethernet
- [x] button
- [x] Wifi2g
- [x] wifi5g
- [x] wifi5g
- [x] mac
- [x] led

### CODE REVIEW

**Comment by Sinbiosis** [comment]
Discussion of work and help was made in [this thread](https://forum.openwrt.org/t/adding-openwrt-support-for-dir-3060-continued/106202/39).

**Comment by musashino205** [comment]
Please copy the contents of the PR comment to commit message of your commit, add Singned-off-by line with your real name and real email address. See: https://openwrt.org/submitting-patches#submission_guidelines
Note: then, please use ```git commit --amend``` and ```git push -f``` instead of opening a new PR

**Comment by f00b4r0** [comment]
If this is an exact copy of already supported hardware, it probably makes more sense to move said hardware DTS to DTSI and have only two stub DTS that only include this DTSI. 
See for instance how it's done in `mt7620a_netgear_ex3x00_ex61xx.dtsi` and associated ex3700, ex6120 and ex6130.

**Comment by RaylynnKnight** [comment]
Couldn't this be handled simply by adding a DEVICE_ALT0_MODEL and DEVICE_ALT0_VARIANT to the existing define for the DIR-3060 since they are essentially the same?

**Comment by Lucky1openwrt** [comment]
some of these devices require the model number embedded in the uboot header
to allow upload so the end images are different
as for the DTSI a better example is the others in it's family the DIR-1950/2640/2660-A1
they even share the same flash layout etc

**Comment by hauke** [comment]
I agree with @RaylynnKnight.
I do not see that it embedded some special serial number somewhere. 
Please fix the formal problems too.

**Comment by Lucky1openwrt** [comment]
did you bother to look in the image for the dir-3060-a1 at starting from byte 64 (0x40)
later boot loaders won't upload the factory image if it's not correct

https://downloads.openwrt.org/snapshots/targets/ramips/mt7621/openwrt-ramips-mt7621-dlink_dir-3060-a1-squashfs-factory.bin

**Comment by hauke** [comment]
I do not see where this pull request creates a different factory image than the one for the DIR-3060.

**Comment by Lucky1openwrt** [comment]
https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=2c9a07ed28e87bbe2d221d48aaa19b04672646a5

**Comment by Lucky1openwrt** [comment]
fyi maybe interested in this update
https://github.com/openwrt/openwrt/commit/f1c9afd801380a05a91d979b475c76cc0a67caae

**Comment by scenario8** [comment]
Can this PR be approved so the DIR-3040 can be supported. Seems like it's a small change to add this device support

**Review by hauke — CHANGES_REQUESTED** [review]
This is the same as the 
````
$ diff -Naur target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts target/linux/ramips/dts/mt7621_dlink_dir-3060-a1.dts 
--- target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts	2023-08-26 16:41:36.710915884 +0200
+++ target/linux/ramips/dts/mt7621_dlink_dir-3060-a1.dts	2023-06-03 14:50:40.398288134 +0200
@@ -6,8 +6,8 @@
 #include <dt-bindings/input/input.h>
 
 / {
-	compatible = "dlink,dir-3040-a1", "mediatek,mt7621-soc";
-	model = "D-Link DIR-3040 A1";
+	compatible = "dlink,dir-3060-a1", "mediatek,mt7621-soc";
+	model = "D-Link DIR-3060 A1";
 
 	aliases {
 		label-mac-device = &gmac0;
@@ -55,7 +55,7 @@
 			label = "white:net";
 			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
 		};
-		
+
 		usb2_white {
 			label = "white:usb2";
 			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
````

The image file definition is also the same:
```
define Device/dlink_dir-3040-a1
  $(Device/dlink_dir-xx60-a1)
  DEVICE_MODEL := DIR-3040
  DEVICE_VARIANT := A1
endef
TARGET_DEVICES += dlink_dir-3040-a1
````
````
define Device/dlink_dir-3060-a1
  $(Device/dlink_dir-xx60-a1)
  DEVICE_MODEL := DIR-3060
  DEVICE_VARIANT := A1
endef
TARGET_DEVICES += dlink_dir-3060-a1
````

Please just define `DEVICE_ALT0_MODEL` and `DEVICE_ALT0_VARIANT` in the `dlink_dir-3060-a1`

**Comment by scenario8** [comment]
Appreciate you looking into this @hauke .

From what I read, D-Link specifically needs the model in the DEVICE_MODEL field, as D-Link checks for it in the uboot header. This is what is done for the DIR-1960/2640/2660-A1 models.

https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=2c9a07ed28e87bbe2d221d48aaa19b04672646a5

I'm new to all this, but I ended up adding the PRs changes in the latest RC and built a version locally for my 3040 which runs great.

**Comment by Lucky1openwrt** [comment]
https://github.com/Lucky1openwrt/openwrt/commit/caa73ab5ca34fdeceb42e8a19e9d55dc33bdd3b1

this is what I would expect it to end up like
it's not tested as I don't have any of the devices
and the added PR has not been tested & needs it
https://github.com/openwrt/openwrt/commit/f1c9afd801380a05a91d979b475c76cc0a67caae

**Comment by hauke** [comment]
If the `DEVICE_MODEL` is checked by the update process a solution like the one proposed by @Lucky1openwrt would be preferred.

**Comment by scenario8** [comment]
Is the a base install build and an upgrade build I can download and test on my DIR-3040 so the PR can be merged?

**Comment by Lucky1openwrt** [comment]
I think we need some communication with Sinbiosis for anything to happen 
or if this is abandoned and someone else submits there own PR

**Comment by Lucky1openwrt** [comment]
the file generated for the example are here 
they may not work at all try at own risk
http://luckys.onmypc.net/openwrt/DIR-30x0/

**Comment by scenario8** [comment]
@Lucky1openwrt  thanks for creating the builds. I took a out of the box DIR-3040 and was able to flash it with the OpenWRT factory build you provided, and also then update it with the sysupgrade build. I successfully ran OpenWRT with the tri band wifi for a few days without issue. From my perspective, these changes can be merged.

**Comment by Lucky1openwrt** [comment]
@scenario8
as you have a device and seems to have the energy and motivation
and if Sinbiosis does not respond maybe you can submit your own PR
I did just ask him on the openwrt forum about this PR i'll see if he responds there

**Comment by scenario8** [comment]
I would like to try and figure out how to do the PR if @Sinbiosis or yourself don't (I thought you created a PR with these changes). Personally, I think OpenWRT is amazing and adding more models is a good thing.

**Comment by Lucky1openwrt** [comment]
I have other modes in the family but not that model
I just made examples of what I would expect it would be on guthub "not submitted as a Pull Request into openwrt tho"
but you or anyone else is welcome to use what's there for a PR 
I do think an owner of the device needs to do this

**Comment by scenario8** [comment]
Makes sense. I’ll create a PR using much of what was proposed. Will report back when ready for review.

**Comment by hauke** [comment]
Please fix the git commit description like mentioned in https://github.com/openwrt/openwrt/pull/11737#issuecomment-1375033315.
Please squash  the commits together.

**Comment by Sinbiosis** [comment]
Okay. I've simplified the dts file to use the dtsi file properly and rebased onto the current commit of master (e1a6c7213f). I've squashed the commits, updated the commit message, and signed as requested. Let me know if I'm missing anything else.

NOTE [informal bug report]: I've noticed an intermittent stability issue. I have three DIR-3040 devices connected together via mesh and, for some reason, the main gateway router will just not respond for about two minutes and then come back. This seems to be random as it does not fall on any set time or minute of the hour, but it kills my internet. I can go days or weeks without issue, then "bye by...hello again". I checked the system logs and kernel logs, but saw nothing unusual. It could have just been because I was building from the head of the master branch and was using an unstable package.

Anyway, here you go folks.

**Comment by wbsmperry** [comment]
Any updates on this, yet?

**Comment by Sinbiosis** [comment]
@wbsmperry @raphaelquati I have heard nothing so far. I'll ping the reviewer, just in case.

@hauke?

File: target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts
Comment (DragonBluep) [inline target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts:54]: `mac-address-increment` support has been removed recently. Please rebase it and use the new nvmem `fixed-layout` instead. And this model has 3 wireless phys, so I believe the MAC addresses are not correct.

Comment (Sinbiosis) [reply target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts]: Okay, I've updated to answer some of these concerns. I'm not sure about the "3 wireless phys, so I believe the MAC addresses are not correct" part. The dts file for the dir-3060 has 2 given "<&macaddr_factory_e000 N>" lines for two wifi cards. Yes there are 3 phys, but I believe that two of them are handled by one wifi device. Plus, this setup works for the DIR-3040 and the three phys all have different MAC addresses.

**Comment by Sinbiosis** [comment]
For this latest commit, I've updated to remove "mac-address-increment = <N>;" lines and use the "nvmem-cells = <&eeprom_factory_0>, <&macaddr_factory_e000 N>" lines. Since I was updating the dts, I also rebased onto the latest master commit (7939df9587).

Once again, please let me know if there are any additional requests.

File: target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts
Comment (DragonBluep) [reply target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts]: > this setup works for the DIR-3040 and the three phys all have different MAC addresses.

Are they the same as the stock firmware? Please add some MAC address explanations in the commit message. This is a good example: f3cdc9f9881796794c06f784a2e4790f5ed75d1f

Comment (Sinbiosis) [reply target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts]: So, I don't recall the original MACs for the stock firmware, but my understanding is that the first wireless interface (WI) is two chips that each serve different bands. WI1 chip1 serves 2G and WI1 chip2 serves 5G. This is also seen in the [deviwiki article for the DIR-3040](https://deviwiki.com/wiki/D-Link_DIR-3060_rev_A1). As such, they have to have the same MAC since it's really the same WI. Here is a table that hopefully answers your question.

| use | address | example |
| --- | --- | --- |
| LAN | label | f4:*:61 |
| WAN | label + 4 | f4:*:65 |
| WI1/2g | label + 1 | f4:*:62 |
| WI1/5g | label + 1 | f4:*:62 |
| WI2/5g | label + 3 | f4:*:64 |

The label MAC address was found in Factory, 0xe000

Again, this was allowed for the DIR-3060 dts file, so I don't see an issue for the DIR-3040 dts file.

If this is enough of an explaination, I can include the table in the commit message.

**Comment by Sinbiosis** [comment]
Updated commit message to include table of MAC addresses.

File: target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts
Comment (DragonBluep) [reply target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts]: Each SSID must have a different MAC, otherwise you will suffer some strange connection issues. I would suggest returning the stock firmware and recording the correct MACs.

Comment (Sinbiosis) [reply target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts]: @Lucky1openwrt  and @iivailo, do either one of you know how to setup the dts file to have a third MAC for the other interface of the abgn+ac radio?

Comment (Lucky1openwrt) [reply target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts]: I don't have any devices with a DBDC interface to check
I do remember the driver use to change a some bytes this differs from the dlink OEM address
I currently expect the driver to alter the MAC address for the 2nd interface based off the 1st 
changing 1st and 4th bytes as below 
`
MAC addresses on OEM firmware:

lan      factory 0xe000   f4:*:*:a8:*:65  (label)
wan      factory 0xe006   f4:*:*:a8:*:68
2.4 GHz  [not on flash]   f6:*:*:c8:*:66
5.0 GHz  factory 0x4      f4:*:*:a8:*:66

The increment of the 4th byte for the 2.4g address appears to vary.
Reported cases:

       5g                 2.4g         increment
 f4:XX:XX:a8:XX:66  f6:XX:XX:c8:XX:66  +0x20
 x0:xx:xx:68:xx:xx  x2:xx:xx:48:xx:xx  -0x20
 x4:xx:xx:6a:xx:xx  x6:xx:xx:4a:xx:xx  -0x20`

https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=b8168f4716ad64863db8a6a2aba24bbff4664ffe
https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=6639623e75b84de5cf4a657be7a80383309e290d

Comment (Sinbiosis) [reply target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts]: @Lucky1openwrt Thank you!

**Comment by Sinbiosis** [comment]
Updated 10_fix_wifi_mac so the 2G side of the abgn+ac radio has a different MAC address from the other interface devices.

File: target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts
Comment (DragonBluep) [inline target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts:48]: Please remove duplicate new line.

File: target/linux/ramips/dts/mt7621_dlink_dir-xx60-a1.dtsi
Comment (DragonBluep) [inline target/linux/ramips/dts/mt7621_dlink_dir-xx60-a1.dtsi:17]: Label `keys: ` is useless.

File: target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac
Comment (DragonBluep) [inline target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac:43]: Looks good to me if it matches the stock firmware behavior. And please sort it alphabetically.

File: target/linux/ramips/dts/mt7621_dlink_dir-xx60-a1.dtsi
Comment (Sinbiosis) [reply target/linux/ramips/dts/mt7621_dlink_dir-xx60-a1.dtsi]: Handled with new amendment

File: target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac
Comment (Sinbiosis) [reply target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac]: Handled with new amendment

**Comment by Sinbiosis** [comment]
Removed empty white space, sorted my entry in 10_fix_wifi_mac, and removed unneeded label

File: target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts
Comment (DragonBluep) [inline target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts:49]: `mediatek,mtd-eeprom = <&factory 0x0>;` can be removed.

File: target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts
Comment (DragonBluep) [inline target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts:56]: `mediatek,mtd-eeprom = <&factory 0x8000>;` can be removed.

File: target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac
Comment (DragonBluep) [inline target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac:75]: Please move it up and place it after case `dlink,dap-x1860-a1`.

Comment (Sinbiosis) [reply target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac]: Handled with new amendment

File: target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts
Comment (Sinbiosis) [reply target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts]: Handled with new amendment

File: target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts
Comment (Sinbiosis) [reply target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts]: Handled with new amendment

**Review by DragonBluep — APPROVED** [review]

**Comment by wbsmperry** [comment]
Do we still need input from Hauke for this to be accepted?

**Comment by Lucky1openwrt** [comment]
looking at this witch looks good
makes the 3060 DTS need an update

**Comment by Sinbiosis** [comment]
I can't seem to do anything on my end about removing @hauke as a reviewer.

**Comment by Sinbiosis** [comment]
Is there any update on this? @Lucky1openwrt @hauke @DragonBluep?

**Comment by Lucky1openwrt** [comment]
FYI there are now changes to the LED's
just a note tho that wan led assignment from 01leds 
is not working on the DIR-1960-A1 atm

**Comment by raenye** [comment]
You should convert the LED entries to the new format.
`color`, `function` and possibly `function-enumerator` instead of `label`. Fix `01_leds` too.

File: target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts
Comment (raenye) [inline target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts:51]:
```suggestion
	/delete-property/ ieee80211-freq-limit;
```

**Review by Lucky1openwrt — APPROVED** [review]

**Comment by Lucky1openwrt** [comment]
up until i just saw my name under reviewers I didn't know I could or I would have already
as is it's as good as the DIR-3060 with better mac addressing 
as for LEDS the one's shown have not been updated on the DIR-3060 but the share ones have same at lest same as 3060
I image the same as the  "/delete-property/ " if needed needs to happen with the DIR-3060 and DIR-853's as well

**Comment by raenye** [comment]
If the frequency limitation is not relevant to many models, it shouldn't be in the dtsi...

**Comment by Lucky1openwrt** [comment]
> If the frequency limitation is not relevant to many models, it shouldn't be in the dtsi

I never know the correct answer to this
do you remove it from dtsi & add it back to the other  4 DTS's
it's the same with the wps gpio & the wan mac address
I'll go with the flow life is to short

**Comment by Lucky1openwrt** [comment]
I had a play with my DIR-1960-A1 just  2x MT7615x4 2.4G & 5G
I tested removing with `/delete-property/ ieee80211-freq-limit;` on both radios
the 2.4G now started asking if you wanted it to be 2.4G or 5G so it needed it
but the 5G was still the same 5G only so maybe not
I don't have a router with a MT7615 DBDC to test myself
but would like to know if you have to select or they just are 2.4G & 5G only ?

**Comment by raenye** [comment]
> the 2.4G now started asking if you wanted it to be 2.4G or 5G so it needed it

Is there a compatible 5G front end on this radio too? so you could have two concurrent 5G bands?

AFAIK MT7615 reads its config (4x4 2.4G, 4x4 5G or DBDC 2x2+2x2) from the EEPROM.

**Comment by Lucky1openwrt** [comment]
it should have all 3 radios there a 2.4G and 2 5G

File: target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts
Comment (Sinbiosis) [reply target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts:51]: I'll test out this change when I get a free moment. Thanks for the suggestion.

**Comment by wbsmperry** [comment]
Does anyone know what can be done to get his approved?  Disappointing that this is still stuck after all of this time.  Im currently running this on my 3040, and have had no issues so far.

**Comment by Lucky1openwrt** [comment]
there is some updated ways to add LED's into the DTS to looks at
and the affects of "/delete-property/ ieee80211-freq-limit;"
as you have a device you could update and try on your device
I would have if I had that model or the 3060

**Comment by Sinbiosis** [comment]
@wbsmperry and @Lucky1openwrt, I got pulled into some extra work stuff and life events so I didn't have alot of time to commit. I should have some free time in the following days to work on updating the LEDs to the new format. But I can't say if this will get approved anytime soon even after my updates. I have no control over the review process.

File: target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts
Comment (hauke) [reply target/linux/ramips/dts/mt7621_dlink_dir-3040-a1.dts:51]: did this change work?

**Comment by hauke** [comment]
Did you test this change: https://github.com/openwrt/openwrt/pull/11737#discussion_r1495064506 ?

**Comment by Sinbiosis** [comment]
Can someone tell me or point me to some documentation about what this "function-enumerator" argument is about?

**Comment by hauke** [comment]
The `function-enumerator ` should be used when you have multiple LEDs with the same function and color. In this case the 2 LEDs for the 5GHy wifi.

It should probably look like this:

````
	wlan2g {
		function = LED_FUNCTION_WLAN_2GHZ;
		color = <LED_COLOR_ID_WHITE>;
		gpios = <&gpio 11 GPIO_ACTIVE_HIGH>;
		linux,default-trigger = "phy0tpt";
	};

	wlan5glb {
		function = LED_FUNCTION_WLAN_5GHZ;
		color = <LED_COLOR_ID_WHITE>;
		function-enumerator = <0>;
		gpios = <&gpio 9 GPIO_ACTIVE_HIGH>;
		linux,default-trigger = "phy1tpt";
	};

	wlan5ghb {
		function = LED_FUNCTION_WLAN_5GHZ;
		color = <LED_COLOR_ID_WHITE>;
		function-enumerator = <1>;
		gpios = <&gpio 5 GPIO_ACTIVE_HIGH>;
		linux,default-trigger = "phy2tpt";
	};
````

I made use of `LED_FUNCTION_WLAN_2GHZ` and `LED_FUNCTION_WLAN_5GHZ` instead, this was integarted in OpenWrt some weeks ago. I think the `tpt` LED trigger are also blinking when there is activity.

**Comment by raenye** [comment]
> I made use of `LED_FUNCTION_WLAN_2GHZ` and `LED_FUNCTION_WLAN_5GHZ` instead, this was integrated in OpenWrt some weeks ago.

@hauke we have quite a few devices using `function-enumerator` for this, exactly because at the time the only available function was `LED_FUNCTION_WLAN`. What is the right procedure to convert them to the new functions (and not break stuff)?

**Comment by Sinbiosis** [comment]
@raenye it seems to me that function-enumerator is just appending the value given to the name of the label that is created for the LED. So you could give a more concise number that might distinguish between the devices. I saw a that "target/linux/ramips/dts/mt7621_rostelecom_rt-fe-1a.dts" and "target/linux/ramips/dts/mt7621_sercomm_s1500.dtsi" and a couple of other dts files used <24> and <50> to distinguish between 2.4 GHz and 5 GHz LEDs before the new "LED_FUNCTION_WLAN_5GHZ" and "LED_FUNCTION_WLAN_2GHZ" functions.

Also, @hauke I've updated the files to use the enumerator and the function and color inputs for the LEDs.

**Comment by Sinbiosis** [comment]
And I tested the new changes on three DIR-3040 A1 devices that have different functions and they all work, LEDs can be configured, and the deleted property doesn't seem to have caused issues.

**Comment by raenye** [comment]
> a couple of other dts files used <24> and <50> to distinguish between 2.4 GHz and 5 GHz LEDs before the new "LED_FUNCTION_WLAN_5GHZ" and "LED_FUNCTION_WLAN_2GHZ" functions.

Indeed, while others use `<2>` and `<5>`. That's what I referred to here:
> we have quite a few devices using `function-enumerator` for this,

My question is whether one is expected to use `function-enumerator` sequentially *now*.

**Comment by hauke** [comment]
> > a couple of other dts files used <24> and <50> to distinguish between 2.4 GHz and 5 GHz LEDs before the new "LED_FUNCTION_WLAN_5GHZ" and "LED_FUNCTION_WLAN_2GHZ" functions.
> 
> Indeed, while others use `<2>` and `<5>`. That's what I referred to here:
> 
> > we have quite a few devices using `function-enumerator` for this,
> 
> My question is whether one is expected to use `function-enumerator` sequentially _now_.

`LED_FUNCTION_WLAN_2GHZ` and `LED_FUNCTION_WLAN_5GHZ` were newly introduced. Before we used `LED_FUNCTION_WLAN` and  `function-enumerator` to distinguish the 2.4GHy and the 5GHy wifi LED. For devices with 2 radios the `function-enumerator` is not needed any more. This device has 3 radios, then we just have to enumerate them.

**Comment by hauke** [comment]
Could you use your real name as author and in the SoB line please. There is currently a discussion about this topic on the mailing list http://lists.openwrt.org/pipermail/openwrt-devel/2024-March/042472.html

File: target/linux/ramips/dts/mt7621_dlink_dir-xx60-a1.dtsi
Comment (hauke) [inline target/linux/ramips/dts/mt7621_dlink_dir-xx60-a1.dtsi:58]: This changes the name of the LED. The name is referenced here: https://github.com/openwrt/openwrt/blob/main/target/linux/ramips/mt7621/base-files/etc/board.d/01_leds#L97

**Comment by wbsmperry** [comment]
@hauke Just wondering which is the requested change that is blocking this from merging?

**Pull request merged**
